AdyenDropIn/Modules/Component/ComponentContainerRouter.swift

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright (c) 2025 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Adyen
+import Foundation
+
+internal protocol ComponentContainerRouterDelegate: AnyObject {
+    func didSubmit(_ data: PaymentComponentData, from component: any PaymentComponent)
+    func didFail(with error: any Error)
+    func didCancel(component: any PaymentComponent)
+}
+
+internal protocol ComponentContainerRouterProtocol {
+    var delegate: ComponentContainerRouterDelegate? { get set }
+    var rootViewController: UIViewController { get }
```

Comment:
Does it need to have `root` in the name?

Reply:
Not necessarily. Just wanna emphasize this is the start view of the router. It can named `viewController`